app/controllers/api/rest_controller.rb

```diff
@@ -35,6 +35,15 @@ def require_trusted_app
       raise ApplicationController::NotAuthorized
     end
 
+    def require_trusted_app_without_current_user
+      has_trusted_app =
+        current_app.present? &&
+          current_user_roles.include?(:trusted)
+      return if has_trusted_app
```

Comment:
This doesn't technically do what it says. _Every_ token will have a `current_app` (identified by the `aud` property). If you want to exclude user tokens, you should additionally verify that there's no `current_user` (identified by the `sub` property).

Reply:
I will rename it to `ensure_trusted_app`
This https://github.com/artsy/convection/blob/main/app/controllers/api/rest_controller.rb#L29 here limits choices of what to name the method for when we only want to check trusted_app without bothering about the sub property.